src/mlpack/core/metrics/hdbscan_metric.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file hdbscanMetric.hpp
+ * @author Sudhanshu Ranjan
+ *
+ * An implementation of the HDBSCAN metric.
+ *  
+ */
+
+#ifndef MLPACK_CORE_HDBSCAN_METRIC_HPP
+#define MLPACK_CORE_HDBSCAN_METRIC_HPP
+
+namespace mlpack {
+namespace metric {
+
+class HdbscanMetric
```

Comment:
Could you add a little documentation to this class please?  To the best of my understanding you are doing a really clever trick here---the vectors that are given to this by HDBSCAN have d_core appended to them.  Whatever the "trick" ends up being, we should document it here because it will make it clear how the class is meant to be used.